mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -299,7 +307,12 @@ class WidgetAdapter(
         if (oldWidget.visibility != widget.visibility && items.any { w -> w.parentId == widget.id }) {
             notifyDataSetChanged()
         } else {
-            notifyItemChanged(position)
+            // update the parent Buttongrid if the updated widget is a button
+            if (widget.type == Widget.Type.Button && widget.parentId != null) {
+                (viewHoldersById[widget.parentId] as? ButtongridViewHolder)?.let { it.updateButton(widget) }
```

Comment:
Instead of this, just find the index of parentId in the list and call `notifyItemChanged(index)` ... this will cause onBind to be called again, where the updated data can be applied.

Reply:
Are you saying: find the parent (buttongrid) and call notifyItemChanged(buttongrid)? No that won't work, because 
- it messes up with click/release because Buttongrid's bind wipes out the tableview and recreates it while the button in the table is still trying to keep track of the mouse down, waiting for it to go up.
- The buttongrid widget doesn't have the updated Button widget state (color, item state)

This is why I made the Widget.buttons mutable, so it can be updated

Reply:
> The buttongrid widget doesn't have the updated Button widget state (color, item state)

Then we need to make it so it does have them ;-)
The flow basically should be
- SSE listener updates widget (here: button grid)
- SSE listener calls back into UI to notify about change
- Adapter uses that callback for rebinding

As for the release handling, I'll need to look into it for a bit before being able to comment properly. My gut feeling is that the view holder needs to keep track whether it's bound to the same or a different widget and if it's the same, keep touch tracking intact.